Sources/App/Settings/SettingsDetailViewController.swift

```diff
@@ -737,6 +737,16 @@ class SettingsDetailViewController: HAFormViewController, TypedRowControllerType
                     }.done {
                         self?.updatePositions()
                     }.cauterize()
+
+                    if vc.shouldOpenAutomationEditor {
+                        vc.navigationController?.dismiss(animated: true, completion: {
+                            Current.sceneManager.webViewWindowControllerPromise.done { controller in
+                                controller.webViewControllerPromise.done { webViewController in
+                                    webViewController.openActionAutomationEditor(actionId: vc.action.ID)
```

Comment:
`then` lets you avoid this double layer of done chaining, e.g. `thing1.then { thing2 }.done { … }` and i think `.then(\.webViewControllerPromise)` may be all you need